File: paa5/paa5/Literal.cs
using System.Text;

namespace paa5
{
	public class Literal
	{
		public bool Negative { get; private set; }

		public int Index { get; private set; }

		public int Weight { get; private set; }

		public Literal(int index, int weight, bool negative)
		{
			this.Index = index;
			this.Weight = weight;
			this.Negative = negative;
		}

		public override string ToString()
		{
			StringBuilder sb = new StringBuilder();
			if (Negative)
			{
				sb.Append('-');
			}
			sb.Append(Index);
			return sb.ToString();
		}
	}
}

File: src/vm/example/sat/Literal.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package vm.example.sat;

/**
 *
 * @author Václav
 */
class Literal {

    public int Negative;

    public int Index;

    public int Weight;

    public Literal(int index, int weight, int negativeBool) {
	this.Index = index;
	this.Weight = weight;
	this.Negative = negativeBool;
    }
}
